Assets/Scripts/checkOneTimeTrigger.cs

```diff
@@ -18,10 +18,11 @@ public class checkOneTimeTrigger : MonoBehaviour {
 	}
 
 	public void UpdateTrigger () {
-		isTriggeredOnce = GameManager.getInstance ().objectTriggered[index];
+		hasBeenTriggeredOnce = GameManager.getInstance ().objectTriggered[index];
```

Comment:
but yes it is kinda the same idea, GameManager keeps track of which object has been triggered, this checkOneTimeTrigger is attached to a triggerObject to keep track whether it itself has been triggered or not. and then u have scripts like SwitchSelectMsg and ActivateTextAtLine that access this script before they give the trigger response(like display UI)